llvm/lib/Target/X86/X86FlagsCopyLowering.cpp

```diff
@@ -258,13 +258,32 @@ bool X86FlagsCopyLoweringPass::runOnMachineFunction(MachineFunction &MF) {
   MRI = &MF.getRegInfo();
   TII = Subtarget->getInstrInfo();
   TRI = Subtarget->getRegisterInfo();
-  MDT = &getAnalysis<MachineDominatorTreeWrapperPass>().getDomTree();
   PromoteRC = &X86::GR8RegClass;
 
   if (MF.empty())
     // Nothing to do for a degenerate empty function...
     return false;
 
+  if (none_of(MRI->def_instructions(X86::EFLAGS), [](const MachineInstr &MI) {
+        return MI.getOpcode() == TargetOpcode::COPY;
```

Comment:
Why don't check `MI.getOperand(0).getReg() == X86::EFLAGS` as well?

Reply:
I don't think that's necessary (or beneficial): if there's a def of eflags from a copy instruction, then the first operand must be eflags.